models/articles-models.js

```diff
@@ -31,5 +31,12 @@ exports.fetchCommentsByArticleId = async (article_id) => {
     `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`,
     [article_id]
   );
-  return result.rows;
+  if (result.rows.length === 0) {
+    return Promise.reject({
+      status: 404,
+      message: "No comments matching the provided article ID",
```

Comment:
This might not always be an error, only if the article is not found do we want to respond with this error but if there are no comments on an existing article that would be a `200`